lightly/transforms/add_grid_transform.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (c) Lightly AG and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+#
+from math import ceil
+from typing import Any, Dict, Tuple
+
+import torch
+from torchvision.transforms.v2 import CenterCrop, ConvertBoundingBoxFormat, Transform
+from torchvision.tv_tensors import BoundingBoxes, BoundingBoxFormat, Mask
+
+from lightly.utils import dependency as _dependency
+
+
+# ignore typing due to Any type used in torchvison.transforms.v2.Transform
+class AddGridTransform(Transform):  # type: ignore[misc]
+    """Implements the naive segmentation into a regular grid from DetCon. [0]
+
+    Input to this transform:
+        Any datastructure containing one or several `torchvision.tv_tensor.BoundingBoxes`
+        and/or `torchvision.tv_tensor.Mask`. Masks should be of size (*, H, W) and
+        BoundingBoxes can be of arbitrary shape.
+
+    Output of this transform:
+        Leaves any images in the data structure untouched, but overwrites any bounding
```

Comment:
Correct, the user has to provide empty/randomly initialized bounding boxes and masks inside the data structure that will be overwritten (maybe this should be made clearer). The v2 transforms should (if possible) work on arbitrary data structures, and return that exact data structure, therefore altering existing bounding boxes/masks is easier than creating them – since we don't know where we should insert them. Also see:

[https://pytorch.org/vision/main/auto_examples/transforms/plot_transforms_getting_started.html#what-do-i-pass-as-input](https://pytorch.org/vision/main/auto_examples/transforms/plot_transforms_getting_started.html#what-do-i-pass-as-input)